astropy/stats/funcs.py

```diff
@@ -744,6 +744,14 @@ def poisson_conf_interval(n, interval='root-n', sigma=1, background=0,
         else:
             conf_interval[0, n == 0] = 0
     elif interval == 'kraft-burrows-nousek':
+        # Deprecation warning in Python 3.9 when N is float, so we force int,
+        # see https://github.com/astropy/astropy/issues/10832
+        if np.isscalar(n):
+            if not isinstance(n, int):
+                raise TypeError('Number of counts must be integer.')
+        elif not issubclass(n.dtype.type, np.integer):
+            raise TypeError('Number of counts must be integer.')
```

Comment:
```suggestion
        if not issubclass(np.array(n).dtype.type, np.integer):
            raise TypeError('Number of counts must be integer.')
```
Could be shortened a bit, I think, or would this be missing some case?

Reply:
Any suggested new verbiage I should use?

Reply:
Oh, you mean the `if` logic, not the error message. Sorry for the misunderstanding.

Please let me know how I can make the check more robust. I see that it handles scalar and ndarray, so I thought this logic would be enough, but I am open to suggestions.

Reply:
No, I think it was OK, just seemed more code than necessary, but I was not sure if _my suggestion_ fully replaced it. But I think it is sufficient, too.

Reply:
Ah, sorry, looks like I missed that you provided a suggestion too. 😬  I guess performance hit of converting a scalar to array is negligible, all things considered, so it boils down to preference? @larrybradley and @hamogu , what say you?